src/FakeItEasy/Sdk/Create.cs

```diff
@@ -36,7 +36,10 @@ public static object Fake(Type typeOfFake)
         /// <returns>A fake object.</returns>
         public static object Fake(Type typeOfFake, Action<IFakeOptions> optionsBuilder)
         {
-            return FakeCreator.CreateFake(typeOfFake, optionsBuilder);
+            Guard.AgainstNull(typeOfFake, nameof(typeOfFake));
+            Guard.AgainstNull(optionsBuilder, nameof(optionsBuilder));

```

Comment:
I don't think these guards belong here but, in folding `DefaultFakeCreatorFacade` into this type, I've brought them along for the ride. I'll address my concerns here in a later PR.